### PR TITLE
gcc: Support Darwin 21 (aka Monterey)

### PR DESCRIPTION
**Pull request by eschnett**

### CODE REVIEW

**Comment by briederer** [comment]
👍

File: var/spack/repos/builtin/packages/gcc/package.py [hunk 1]
```diff
@@ -258,6 +258,8 @@ class Gcc(AutotoolsPackage, GNUMirrorPackage):
             # (otherwise __OSX_AVAILABLE_STARTING ends up undefined)
             patch('https://raw.githubusercontent.com/Homebrew/formula-patches/b8b8e65e/gcc/9.2.0-catalina.patch',
                   sha256='0b8d14a7f3c6a2f0d2498526e86e088926671b5da50a554ffa6b7f73ac4f132b', when='@9.2.0')
+        if macos_version() >= Version('12'):
+            patch('darwin/patch-darwin21-support.diff', when='@11:')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/gcc/package.py:262]: Does this work too:
```suggestion
        patch('darwin/patch-darwin21-support.diff', when='@11: platform=darwin os=monterey')
```
?

**Review by alalazo — CHANGES_REQUESTED** [review]
Left a minor comment. Will try building GCC later. Is it possible to reuse a remote patch?

File: var/spack/repos/builtin/packages/gcc/package.py (hunk 1 above)
Comment (eschnett) [reply var/spack/repos/builtin/packages/gcc/package.py:262]: This should work too.

**Comment by eschnett** [comment]
I am not aware of a remote patch, although either MacPorts or Homebrew might have one that we could reuse.

**Comment by briederer** [comment]
In case it is helpful. Found these in the [Homebrew repo for gcc](https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/gcc.rb):

General patch ([L17-L20](https://github.com/Homebrew/homebrew-core/blob/7e1d5046945620245be075f23caf2d23646ee3e3/Formula/gcc.rb#L17-L20)): https://github.com/iains/gcc-darwin-arm64/commit/20f61faaed3b335d792e38892d826054d2ac9f15.patch?full_index=1

And a patch which I honestly don't get what it does 😅 ([L62-L67](https://github.com/Homebrew/homebrew-core/blob/7e1d5046945620245be075f23caf2d23646ee3e3/Formula/gcc.rb#L62-L67)): https://gcc.gnu.org/git/?p=gcc.git;a=patch;h=fabe8cc41e9b01913e2016861237d1d99d7567bf

**Comment by eschnett** [comment]
The general Homebrew patch above looks good. We could use it instead of mine.

**Comment by alalazo** [comment]
> The general Homebrew patch above looks good. We could use it instead of mine.

Yeah, let's go ahead with these two minor modifications and merge this :slightly_smiling_face:

**Comment by sethrj** [comment]
Superseded by #28662

**Pull request closed without merging**